js/webform_civicrm_forms.js

```diff
@@ -131,7 +131,21 @@ var wfCivi = (function ($, D) {
           } else {
             $(':input', this).not(':radio, :checkbox, :button, :submit, :file, .form-file').each(function() {
               if (this.id && $(this).val() != '') {
-                $(this).val('');
+                //Set default value if it is specified in component settings.
+                if (typeof defaults != "undefined" && defaults.hasOwnProperty(name)) {
+                  if ($el.hasClass('webform-component-date')) {
+                    var date = defaults[name].split('-');
+                    $el.find('select.year, input.year').val(+date[0]);
+                    $el.find('select.month').val(+date[1]);
+                    $el.find('select.day').val(+date[2]);
+                  }
```

Comment:
I don't understand how this would work. We're in a loop of all `$(':input', this).not(':radio, :checkbox, :button, :submit, :file, .form-file')` but acting on a wrapper div around other form elements.

Reply:
Moved `$el` update to little above so that it is modified in the div loop initiated in the `resetfields()` function rather than inside `$(':input', this)`. 

Confirmed the updated changes and it too works fine on my testing.